packages/studio-plugin/src/createStudioStylingPlugin.ts

```diff
@@ -0,0 +1,58 @@
+import { PluginOption } from "vite";
+import upath from "upath";
+import { fileURLToPath } from "url";
+const currentDir = upath.dirname(fileURLToPath(import.meta.url));
+const packagesDir = upath.resolve(currentDir, "../..");
+
+/**
+ * Adds the ?inline query parameter to all CSS imports
+ * unless they are CSS from Studio, which is determined
+ * from the path of its importer. This prevents user styling
+ * from affecting Studio UI.
+ */
+export default function createStudioStylingPlugin(): PluginOption {
+  return {
+    name: "yext-studio-style-plugin",
+    enforce: "pre",
+    async resolveId(id, importer) {
+      if (!importer) {
```

Comment:
when would the `importer` not be defined?

Reply:
I think it's only the case for the entry point of the plug-in: https://rollupjs.org/plugin-development/#resolveid